### PR TITLE
Fix TiKV panic when aggregation expr type is not valid

### PR DESCRIPTION
**Pull request by breeswish**
Signed-off-by: Breezewish <breezewish@pingcap.com>

###  What have you changed?

This PR protects TiKV from panic in scenarios like https://github.com/tikv/tikv/issues/5404

###  What is the type of the changes?

- Bugfix (a change which fixes an issue)

###  How is the PR tested?

Manually, using a TiDB commit that will produce invalid expr type:

```
$ mysql -vvv --host 127.0.0.1 --port 4000 -u root tpch < queries/1.sql
--------------
select
	l_returnflag,
	l_linestatus,
	sum(l_quantity) as sum_qty,
	sum(l_extendedprice) as sum_base_price,
	sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
	sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
	avg(l_quantity) as avg_qty,
	avg(l_extendedprice) as avg_price,
	avg(l_discount) as avg_disc,
	count(*) as count_order
from
	lineitem
where
	l_shipdate <= date_sub('1998-12-01', interval 108 day)
group by
	l_returnflag,
	l_linestatus
order by
	l_returnflag,
	l_linestatus
--------------

ERROR 1105 (HY000) at line 4: other error: [components/tidb_query/src/aggr_fn/impl_sum.rs:49]: Unexpected return field type LongLong
```

### CODE REVIEW

**Comment by breeswish** [comment]
@sticnarf @lonng @mapleFU  PTAL, thanks

File: components/tidb_query/src/aggr_fn/impl_avg.rs [hunk 1]
```diff
@@ -34,14 +34,8 @@ impl super::AggrDefinitionParser for AggrFnDefinitionParserAvg {
...
 
-        // AVG outputs two columns.
-        out_schema.push(
-            FieldTypeBuilder::new()
-                .tp(FieldTypeTp::LongLong)
-                .flag(FieldTypeFlag::UNSIGNED)
-                .build(),
-        );
-        out_schema.push(aggr_def.take_field_type());
+        let out_column_2 = aggr_def.take_field_type();
```
Comment (sticnarf) [inline components/tidb_query/src/aggr_fn/impl_avg.rs]: `out_column_2` -> `sum_field_type`?

File: components/tidb_query/src/aggr_fn/impl_avg.rs [hunk 2]
```diff
@@ -34,14 +34,8 @@ impl super::AggrDefinitionParser for AggrFnDefinitionParserAvg {
...
+        let out_column_2_et = box_try!(EvalType::try_from(out_column_2.as_accessor().tp()));
```
Comment (sticnarf) [inline components/tidb_query/src/aggr_fn/impl_avg.rs]: `out_column_2_et` -> `sum_eval_type`?

**Comment by sticnarf** [comment]
By the way, as a bug fix, I think there should be a corresponding test.

**Pull request merged**